Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +390,178 @@ namespace Tensile
+                    return std::numeric_limits<double>::quiet_NaN();
+
+                size_t PowerMidValueIndex = m_powerValues.size() / 2;
+                std::nth_element(m_powerValues.begin(), m_powerValues.begin()+PowerMidValueIndex, m_powerValues.end());
+                return static_cast<double> (m_powerValues[PowerMidValueIndex]);
+            }
+           
+            throw std::runtime_error("Can't read Gfx value that wasn't requested: ");
+        }
+        
+        double HardwareMonitor::getAveragePower()
+        {
+            assertNotActive();
+
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+
+            if (m_powerValues.size() != 0)
+            {
+                if(m_powerValues[0] == std::numeric_limits<uint16_t>::max())
+                    return std::numeric_limits<double>::quiet_NaN();
+                    
+                return static_cast<double> (std::accumulate(m_powerValues.begin(), m_powerValues.end(), 0.0) / m_powerValues.size()) ;
+            }
+            
+            throw std::runtime_error("Can't read Power value that wasn't requested: ");
+        }
+        
+        double HardwareMonitor::getMedianHotSpotTemperature()
+        {
+            assertNotActive();
+            
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+
+            if (m_temperatureValues.size() != 0)
+            {
+                if(m_temperatureValues[0] == std::numeric_limits<uint16_t>::max())
+                    return std::numeric_limits<double>::quiet_NaN();
+                    
+                size_t TemperatureMidValueIndex = m_temperatureValues.size() / 2;
+                std::nth_element(m_temperatureValues.begin(), m_temperatureValues.begin()+TemperatureMidValueIndex, m_temperatureValues.end());
+                return static_cast<double> (m_temperatureValues[TemperatureMidValueIndex]);
+            }
+            
+            throw std::runtime_error("Can't read temperature value that wasn't requested: ");
+        }
+        
+        double HardwareMonitor::getAverageHotSpotTemperature()
+        {
+            assertNotActive();
+            
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+
+            if (m_temperatureValues.size() != 0)
+            {
+                if(m_temperatureValues[0] == std::numeric_limits<uint16_t>::max())
+                    return std::numeric_limits<double>::quiet_NaN();
+                    
+                return static_cast<double> (std::accumulate(m_temperatureValues.begin(), m_temperatureValues.end(), 0.0) / m_temperatureValues.size()) ;
+            }
+            
+            throw std::runtime_error("Can't read temperature value that wasn't requested: ");
+        }
+        
+        void HardwareMonitor::logMinMaxMedianAverage()
```

Comment:
Are we using this func?

Reply:
currently not using this function but will eventually use this in future with new switch added otherwise it will be test time hit.